### ISSUE TITLE
Investigate parenthesis-only Render A/B edits classified as structural

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#7699]
## Observation

During the next #2376 spelling slice after #7688, Render A/B classified two
parenthesis-only edits as `structural`, rather than `paren-equivalent`.
Both sides parse without syntax diagnostics and remain semantically valid.

The raw sensor verdict is retained; this report does not silently reclassify
its results or claim whole-program equivalence.

## Real cases

`System.Data.SqlTypes.SqlBytes.get_MaxLength`:

```diff
-return _rgbBuf is null ? ((long)-1) : ((long)_rgbBuf.Length);
+return _rgbBuf is null ? (long)-1 : (long)_rgbBuf.Length;
```

`System.Number.WideDropLowDigits<TValue>`:

```diff
-sticky |= ((uint)WideDivideByTen<TValue>(ref high, ref low) > (uint)0);
+sticky |= (uint)WideDivideByTen<TValue>(ref high, ref low) > (uint)0;
```

Input: frozen .NET 11 RC1 `System.Data.Common.dll` and
`System.Private.CoreLib.dll`. Base product:
`78baa257041f78db2b24af6235ae7e433ea49499`.

## Reproduction and evidence

`tools/DecompilerHarness/RenderAbSensor.cs` parses the body in an async method
shell, removes `ParenthesizedExpressionSyntax` with a `CSharpSyntaxRewriter`,
then calls `SyntaxFactory.AreEquivalent(..., topLevel: false)`.

A separate file-based app using the same `Microsoft.CodeAnalysis.CSharp`
5.9.0 reproduces `AreEquivalent == false` for both pairs. After stripping,
the node/token kind and token-text preorder sequences match (85 entries for
SqlBytes; 124 for WideDropLowDigits). That sequence comparison is diagnostic
evidence, not a replacement grammar/semantic oracle. The precise reason for
Roslyn's non-equivalence remains to be isolated.

The fixed 10,911-method population reported 99 changes:
95 `paren-equivalent`, these two `structural`, and two unrelated pre-existing
`unparsed` methods. No valid-to-invalid transition was observed.

Preserved in session `e9a110eb-c81b-44c5-b1cd-6c640710ca11`:

- `files/precedence-library-diffs/manifest.json`
- `files/precedence-library-diffs/0001.structural-diff.json`
- `files/precedence-library-diffs/0002.structural-diff.json`
- `files/precedence-library-audit.txt`
- `files/precedence-audit.cs` (the independent observation script)

## Next focused work

Minimize the Roslyn discrepancy and determine whether it is an equivalence
API limitation, a rewriter interaction, or a meaningful syntax distinction
that the current diagnostic misses. Preserve a regression for the established
classification contract. Do not substitute text/token equality for the
sensor's syntactic comparison without a justified grammar-level contract.
